experimental/iterators/include/iterators/Dialect/Iterators/IR/IteratorsOps.td

```diff
@@ -265,6 +266,29 @@ def Iterators_ReduceOp : Iterators_Op<"reduce",
   }];
 }
 
+def Iterators_TabularViewToStreamOp : Iterators_Op<"tabular_view_to_stream",
+    [AllMatch<["$input.getType().cast<TabularViewType>().getColumnTypes()",
+               "$result.getType().cast<StreamType>().getElementType()"
+               "    .cast<LLVM::LLVMStructType>().getBody()"],
+              "the field types of the LLVM struct in the result stream must "
+              "match the column types of the tabular view given as input">]> {
+  let summary = "Extracts the tuples from a tabular view as LLVM structs";
+  let description = [{
+    Produces a stream of LLVM structs from a given `tabular_view` (i.e., "scans"
+    the given `tabular_view`). Each LLVM struct represents one row, and the op
+    produces all rows in ascending order.
+
+    Example:
+    ```mlir
+    %stream = "iterators.tabular_view_to_stream"(%view)
+      : (!iterators.tabular_view<!t1, ..., !tn>)
+        -> !iterators.stream<!llvm.struct<(!t1, ..., !tn)>>
```

Comment:
The dependency on LLVM here is very unfortunate.

Reply:
Yes! Added TODO in latest commit.